pandokia/sql/mysql.sql

```diff
@@ -84,8 +84,8 @@ CREATE TABLE result_scalar (
                 -- 0 or 1 indicating whether this test is a chronic problem
 	);
 
-CREATE UNIQUE INDEX result_scalar_test_identity 
-	ON result_scalar ( test_run, project, host, test_name, context );
+CREATE UNIQUE INDEX result_scalar_key_id 
+	ON result_scalar ( key_id );
```

Comment:
OK, I know this fixes the truncated-key-blob issue, but I don't understand how the new key works.  Is result_scalar_key_id just some unigue integer chosen for us?  How does it map to test_run+project+host+test_name+context ?

Or, put another way, why are we not just using those 5 fields as the indices?

Reply:
result_scalar_key_id is enforcing the key_id in result_scalar table to be unique and uses it as index.
The now unique index, key_id, is also the identifier, aka primary key, of the result_scalar table, and that is how key_id maps to test_run+project+host+test_name+context.
The key_id also maps to the fields on other tables, such as result_tda, result_tra and result_log.

Indices should be unique and the original five fields (individually) are not unique.
That is why the original design of the schema put all five fields together to make a unique index but the database engine has a size limit in the length of the index so we can not use all five fields together.